src/stream/src/common/table/state_table.rs

```diff
@@ -832,19 +836,13 @@ impl<S: StateStore> StateTable<S> {
             }
         }
         if let Some(range_end_suffix) = range_end_suffix {
-            let range_begin_suffix = if let Some(ref last_watermark) = self.last_watermark {
```

Comment:
@Little-Wallace told me to write this part of code. If you want to remove it, I recommend to discuss with him.

Reply:
We only need to remove all data before some watermark such as (-∞, t]

Reply:
> We only need to remove all data before some watermark such as (-∞, t]

You said that you preferred a lower bound though your algorithm can work without lower bound.